langchain/src/callbacks/manager.ts

```diff
@@ -130,66 +130,75 @@ class BaseRunManager {
  * Manages callbacks for retriever runs.
  */
 export class CallbackManagerForRetrieverRun
-  extends BaseRunManager
-  implements BaseCallbackManagerMethods
-{
-  getChild(tag?: string): CallbackManager {
-    // eslint-disable-next-line @typescript-eslint/no-use-before-define
-    const manager = new CallbackManager(this.runId);
-    manager.setHandlers(this.inheritableHandlers);
-    manager.addTags(this.inheritableTags);
-    manager.addMetadata(this.inheritableMetadata);
-    if (tag) {
-      manager.addTags([tag], false);
+    extends BaseRunManager {
```

Comment:
Can you revert changes in this file?

Reply:
These changes were to ensure we have a CallbackManagerForChainRun (instead of a generic CallbackManager) to use in line 92

     const response = await this.llmChain._call({ question, k }, runManager?.getChainChild());
     
Do you have a suggestion how to solve this differently?